python/graphstorm/graphpartitioning/metadata_schema.py

```diff
@@ -0,0 +1,799 @@
+        else:
+            if not os.path.isfile(file_name):
+                raise ValueError(f"Metadata file does not exist. {file_name}")
+            self._metadata_path = file_name
+
+        if self._basedir is None:
+            self._basedir = os.getcwd()
+
+        with open(self._metadata_path, "r", encoding="utf-8") as handle:
+            self._data = json.load(handle)
+
+    def _check_ifexists(self, data_files, src_type):
+        """Helper function to verify the presence of a set of filenames.
+
+        Argument
+        --------
+        data_files: list of str
+            A set of file names, either in absolute or in relative formats.
+        src_type: str
+            Either node type or edge type from the input graph.
+        """
+        if not isinstance(data_files, list):
+            raise ValueError(
+                f"Incorrect value encountered when iterating over files for {src_type}."
+            )
+
+        for filename in data_files:
+            if not os.path.isabs(filename):
+                if not os.path.isfile(os.path.join(self._basedir, filename)):
+                    raise FileNotFoundError(
+                        f"File: {os.path.join(self._basedir, filename)} does not exist"
+                    )
+            else:
+                if not os.path.isfile(filename):
+                    raise FileNotFoundError(f"File: {filename} does not exist.")
+
+    def init(self, file_path, base_dir, init_maps=True):
+        """Initialization function of this class to create all the data
+        structures used during typical processing of this pipeline.
+
+        Argument
+        --------
+        file_path: str
+            Path of the metadata schema file of the input dataset. It can be
+            either in absolute or relative formats.
+        base_dir: str
+            Absolute path of the directory in which the metadata schema file
+            is expected to be present. If not provided then current workding
+            directory is used instead to locate the metadata schema file.
+        init_maps: bool
+            Boolean flag to indicate whether to build all the necessary data
+            structures during initialization. Otherwise the user of this class
+            is exptected to invoke the appropriate initialization function.
+        """
+        self._load_json(file_path, base_dir)
+        if init_maps:
+            self.init_maps()
+            self.validate_metadata()
+
+    def init_maps(self):
+        """Initialization function to build all the necessary data structures
+        used during typical processing of the pipeline.
+        """
+        self.init_global_nid_offsets()
+        self.init_global_eid_offsets()
+        self.init_ntype_maps()
+        self.init_etype_maps()
+        self.init_ntype_features()
+        self.init_etype_features()
+        self.init_etype_files()
+        self.init_ntype_feature_files()
+        self.init_etype_feature_files()
```

Comment:
Why are all these functions public if the caller is not supposed to have access to them? The fact that they don't take arguments is an indication that they should just be called in the constructor.

Reply:
We can definitely make them private. Once they are made so, either we can call all of them from the constructor or call them individually from the derived/child class to enable modular testing. As indicated earlier, if we use child/derived class it comes with its own complications which needs to include some funny code in the base class constructor.